store/postgres/src/chain_store.rs

```diff
@@ -593,30 +594,57 @@ mod data {
             &self,
             conn: &PgConnection,
             hash: &BlockHash,
-        ) -> Result<Option<BlockNumber>, StoreError> {
+        ) -> Result<Option<(BlockNumber, Option<String>)>, StoreError> {
+            #[derive(Deserialize)]
+            struct Block {
+                /// timestamp's representation depends the blockchain::Block implementation, on
+                /// ethereum this is a U256 but on different chains it will most likely be different.
+                /// Parsing as string should be safe for most cases as serde should convert.
+                timestamp: String,
+            }
+            #[derive(Deserialize)]
+            struct Data {
+                block: Block,
+            }
+
             let number = match self {
                 Storage::Shared => {
                     use public::ethereum_blocks as b;
 
                     b::table
-                        .select(b::number)
+                        .select((b::number, b::data))
```

Comment:
Instead of getting the whole block from the db, we should just get the timestamp. In SQL, that would look like `select data->'block'->>'timestamp' from chain41.blocks limit 1;` I believe you can get that through Diesel by doing `select(b::number, sql("data->'block'->>'timestamp'"))` though it might require some `QueryableByName` struct.

There's a difference between the JSON structure of how a full and a light block is stored, so you'll have to use `coalesce(data->'block'->>'timestamp', data->>'timestamp')` (see the comment in `Storage::blocks`) 

A while ago we tried to make the chain store dumb wrt the JSON structure of `data`, but I don't see how we can maintain that for this without dragging tens of kB around for each lookup like this.